src/track/beats.cpp

```diff
@@ -21,10 +21,48 @@ struct BeatGridV1Data {
 
 constexpr double kEpsilon = 0.01;
 
+// The amount that `Beats::tryAdjustTempo()` changes the last marker's BPM by.
+constexpr double kBpmAdjustStep = 0.01;
+
+int roundBeatCountToFullBar(int numBeats, int beatsPerBar) {
+    const int numExcessBeats = (numBeats % beatsPerBar);
+    if (numExcessBeats < (beatsPerBar / 2)) {
+        return numBeats - numExcessBeats;
+    }
+
+    return numBeats + beatsPerBar - numExcessBeats;
+}
+
+int calculateBeatsTillNextMarker(const mixxx::Beats::ConstIterator& it,
```

Comment:
The function name is misleading. How about: 
`int beatsTillBarNearPosition()`

Reply:
It looks like there is a need for "calculateBeatsTillNextMarker()" this should however be independent form a Bar.